state-network.md

```diff
@@ -255,53 +269,297 @@ sedes      := Container(enrs: List[byte_list, max_length=32], payload: byte_list
 
-We encode proofs as a stream of `(path, node)` pairs which have been sorted in preorder traversal ordering.  We can save some bytes by de-duplicating the `path` components since each path will tend to have a common prefix with the previous path.
+In this context the term "neighborhood gossip" refers to a DHT node
+re-distributed data that they were interested in to the DHT nodes nearby who
+are also interested.
+
+The phrase "initialization of XXX gossip" refers to finding the DHT nodes that
+are responsible for XXX and offering the data to them.
+
+
+#### Stage 1: Data Ingress
+
+
+The first stage of gossip is performed by a bridge node. Each time a new block
+is added to their view of the chain, a set of merkle proofs which are all
+anchored to `Header.state_root` is generated which contains.
+
+- Account trie Data:
+    - All of the intermediate and leaf trie nodes from the account trie necessary to prove new and modified accounts.
+    - All of the intermediate and leaf trie nodes from the account trie necessary for exclusion proofs for deleted accounts.
+- Contract Storage trie data:
+    - All of the intermediate and leaf trie nodes from each contract storage trie necessary to prove new and modified storage slots.
+    - All of the intermediate and leaf trie nodes from each contract storage trie necessary for exclusion proofs for zero'd storage slots.
+- All contract bytecode for newly created contracts
+
+> TODO: Figure out language for defining which trie nodes from this proof the bridge node must initialize gossip.
+
+> TODO: Determine mechanism for contract code.
+
+
+#### Stage 2A: Neighborhood Trie Node Gossip
+
+
+When individual trie nodes are gossiped they will be transmitted as both the trie node itself, and the additional proof nodes necessary to prove the trie node against a recent state root.
+
+The receiving DHT node will perform *neighborhood* gossip to nearby nodes from their routing table.
+
+
+#### Stage 2B: Recursive Trie Node Gossip
+
+When individual trie nodes are gossiped, the receiving node is responsible for initializing gossip for other trie nodes contained in the accompanying proof.
+
+This diagram illustrates the proof for the trie node under the path `0011`.  
+
+```
+0:                           X
+                            / \
+                          /     \
+                        /         \
+                      /             \
+                    /                 \
+                  /                     \
+                /                         \
+1:             0                           1
+             /   \
+           /       \
+         /           \
+2:      0             1
+       / \
+      /   \
+3:   0     1
+          / \
+4:       0  (1)
+
+```
+
+Note that it also includes the intermediate trie nodes under the paths:
+
+- `000`
+- `001`
+- `00`
+- `01`
+- `0`
+- `1`
+- `X`
```

Comment:
I still find this way of talking about the proofs confusing. Let's talk about the bodies of the nodes and the hashes of the nodes. For example, only the hash of the node at `001` is required to prove inclusion. But the body of the node at `00` *includes* the hash of `001`. So we only really need the bodies of the trie nodes at `X`, `0`, `00`, `001`, and `0011`. All the rest of the data is redundant, AFAICT.